### PR TITLE
Address overhead concerns when sending 100% tx

### PR DESCRIPTION
**Pull request by ale-cota**
We currently have inconsistent messaging in docs:
- on the [Performance at Scale](https://github.com/getsentry/sentry-docs/pull/6427) page we encourage people to send 100% of transactions no matter what
- in SDK docs we caution people about needing to lower the sample rate when going to production. 

We need to acknowledge that there is the “ideal” situation, where sending all transactions is possible and this enables Performance to work best, but there are some scenarios where this just isn’t realistic. For some industries where latency is highly important or even just a large enough scale, it’s not an option. Adding a note about this for clarity.

Related to https://github.com/getsentry/sentry-docs/pull/6427

### CODE REVIEW

File: src/docs/product/performance/performance-at-scale/benefits-performance-at-scale.mdx [hunk 1]
```diff
@@ -21,7 +21,9 @@ Sentry's [retention priorities](/product/performance/performance-at-scale/#reten
...
 
-Our retention priorities work best when you send us as much data as you can. The more a complete picture of your application we have, the better decisions we make on your behalf to retain data and detect and create new issues. If feasible, send us 100% of your data (that is, set your [tracesSampleRate](/platform-redirect/?next=/performance/) in the SDK to 1.0). If you can't do so initially, the system will operate on what it can see and when you purchase more transaction volume we'll adapt automatically. As your transaction volume goes up, the per-transaction pricing goes down accordingly, so you can truly scale affordably.
+Our retention priorities work best when you send us as much data as you can. The more a complete picture of your application we have, the better decisions we make on your behalf to retain data and detect and create new issues. 
+
```
Comment (lizokm) [inline src/docs/product/performance/performance-at-scale/benefits-performance-at-scale.mdx:25]:
```suggestion

Sentry's retention priorities work best when you send high volumes of data. The more data you send, the more accurate a picture of your application we get. This makes it possible to make more precise data retention decisions and to detect and create new issues.

```

File: src/docs/product/performance/performance-at-scale/index.mdx [hunk 2]
```diff
@@ -89,7 +89,9 @@ Sentry offers SDK sampling configuration options: using `tracesSampleRate` enabl
...
 
-In order to send us 100% of your transaction events, set your [tracesSampleRate](/platform-redirect/?next=/performance/) in the SDK to 1.0. If you can't do so initially, the system will operate on what it can see and when you purchase more transaction volume, we'll adapt automatically. As your transaction volume goes up, the per-transaction pricing goes down accordingly, so you can truly scale affordably. Learn more in [Benefits of Performance at Scale](/product/performance/performance-at-scale/benefits-performance-at-scale/).
+In order to send us 100% of your transaction events, set your [tracesSampleRate](/platform-redirect/?next=/performance/) in the SDK to 1.0. When facing overhead concerns, for example with high volume back-end applications, you may want to consider setting a lower value, or switching to using to selectively sample and filter your transactions, based on contextual data.
+
+As your transaction volume goes up, the per-transaction pricing goes down accordingly, so you can truly scale affordably. Learn more in [Benefits of Performance at Scale](/product/performance/performance-at-scale/benefits-performance-at-scale/).
```
Comment (lizokm) [inline src/docs/product/performance/performance-at-scale/index.mdx:94]:
```suggestion
As your transaction volume goes up, the per-transaction pricing will go down accordingly, making it possible to scale affordably. Learn more in [Benefits of Performance at Scale](/product/performance/performance-at-scale/benefits-performance-at-scale/).
```

**Review by lizokm — APPROVED** [review]
Made a few re-wording suggestions :)

File: src/docs/product/performance/performance-at-scale/benefits-performance-at-scale.mdx [hunk 3]
```diff
@@ -19,9 +19,13 @@ Sentry's [retention priorities](/product/performance/performance-at-scale/#reten
...
+Sentry's retention priorities work best when you send high volumes of data. The more data you send, the more accurate a picture of your application we get. This makes it possible to make more precise data retention decisions and to detect and create new issues.
+
+If possible, send us 100% of your data by setting your [tracesSampleRate](/platform-redirect/?next=/performance/) in the SDK to 1.0. Otherwise, the system will operate based on what it can see and will adapt automatically if you purchase more [Performance Units](/product/accounts/pricing/#performance-pricing).
+
+If you want to offset overhead, (with high-volume back-end applications, for example), you can set a lower value, or switch to using selective sampling and start filtering your transactions based on contextual data.
```
Comment (ale-cota) [inline src/docs/product/performance/performance-at-scale/benefits-performance-at-scale.mdx:28]:
```suggestion
If you want to offset overhead, (with high-volume back-end applications, for example), you can set a lower value, or switch to sampling selectively by using the <PlatformIdentifier name="traces-sampler" /> to filter your transactions based on contextual data.
```

File: src/docs/product/performance/performance-at-scale/index.mdx [hunk 4]
```diff
@@ -89,7 +89,9 @@ Sentry offers SDK sampling configuration options: using `tracesSampleRate` enabl
...
 
-In order to send us 100% of your transaction events, set your [tracesSampleRate](/platform-redirect/?next=/performance/) in the SDK to 1.0. If you can't do so initially, the system will operate on what it can see and when you purchase more transaction volume, we'll adapt automatically. As your transaction volume goes up, the per-transaction pricing goes down accordingly, so you can truly scale affordably. Learn more in [Benefits of Performance at Scale](/product/performance/performance-at-scale/benefits-performance-at-scale/).
+To send 100% of your transaction events, set your [tracesSampleRate](/platform-redirect/?next=/performance/) in the SDK to 1.0. If you want to offset overhead, (with high-volume back-end applications, for example), set a lower value, or switch to using selective sampling and start filtering your transactions based on contextual data.
```
Comment (ale-cota) [inline src/docs/product/performance/performance-at-scale/index.mdx:92]:
```suggestion
To send 100% of your transaction events, set your [tracesSampleRate](/platform-redirect/?next=/performance/) in the SDK to 1.0. If you want to offset overhead, (with high-volume back-end applications, for example), set a lower value, or switch to sampling selectively by using the <PlatformIdentifier name="traces-sampler" /> to filter your transactions based on contextual data.
```

**Review by lizokm — APPROVED** [review]
LGTM

**Review by smeubank — APPROVED** [review]

**Pull request merged**